apps/total-typescript/src/module-builder/upload-file.ts

```diff
@@ -0,0 +1,57 @@
+import axios from 'axios'
+import uuid from 'shortid'
+import fileExtension from 'file-extension'
+
+const SIGNING_URL = `/api/aws/sign-s3`
+
+export async function uploadToS3({
+  fileType,
+  fileContents,
+  onUploadProgress = () => {},
+}: {
+  fileType: string
+  fileContents: File
+  onUploadProgress: (progressEvent: {loaded: number; total: number}) => void
+}) {
+  const presignedPostUrl = await getPresignedPostUrl(
+    fileType,
+    fileContents.name,
+  )
+
+  console.log('presignedPostUrl', presignedPostUrl)
+
+  await axios.put(presignedPostUrl.signedUrl, fileContents, {
+    headers: {'Content-Type': 'application/octet-stream'},
+    onUploadProgress,
```

Comment:
axios has `onUploadProgress` because it uses xhr under the hood. You can't get progress with fetch.